pkg/api/mlflow/service/artifact/storage/gs.go

```diff
@@ -25,10 +27,22 @@ type GS struct {
 }
 
 // NewGS creates new GC instance.
-func NewGS(ctx context.Context, _ *config.ServiceConfig) (*GS, error) {
+func NewGS(ctx context.Context, config *config.ServiceConfig) (*GS, error) {
+	if config.GSEndpointURI != "" {
+		client, err := storage.NewClient(
+			ctx, option.WithEndpoint(config.GSEndpointURI), option.WithoutAuthentication(),
```

Comment:
we should build a list of options here instead, and then pass it `storage.NewClient` below, so it's always called in the same place — your current approach duplicates quite a bit of code

Reply:
yes, you are right. done.